package.json

```diff
@@ -46,6 +46,7 @@
     "watch:doc:react": "env BABEL_ENV=transpilation styleguidist server --config docs/styleguide.config.js",
     "watch:doc:kss": "nodemon --ext styl,md --watch stylus --exec 'yarn build:doc:kss && http-server build/styleguide -p 4242'"
   },
+  "sideEffects": false,
```

Comment:
Did you find some guidelines to know when it is acceptable to use `sideEffects: false`? I've searched and didn't find some clear rules. But, IIRC, styles have no `export` and are viewed as as side-effect by webpack. So, we should probably exclude them here.

Reply:
> Did you find some guidelines to know when it is acceptable to use sideEffects: false? 

I have read https://github.com/webpack/webpack/tree/master/examples/side-effects. It is not acceptable when your code has sideEffects (inserting CSS files for example) but: 

* We are not in this case after transpilation
* We could be in this case without transpilation but I think it can be ignored. If we tell webpack that stylus files have side effects, it means that our component files have side effects, and we cannot optimize for component usages. We should strive for our stylus files for components not to have side effects (global CSS rules).

---

Longer answer :

### What does `sideEffects: false` tell webpack

Often, we have cases where we have `index.js` reexporting components : 

lib/A.jsx
```js
import styles from './A.styl'
export default () => <span>Component A</span>
```

lib/B.jsx
```js
import styles from './B.styl'
export default () => <span>Component B</span>
```

lib/index.js
```js
export { default as A } from './A'
export { default as B } from './B'
```

app/index.js
```js
import { A } from 'lib'
```

Here since we do not import B in app/index.js, the export is flagged as unnecessary in webpack: 

webpack::lib/index.js
```js
export { default as A } from './A'
/* harmony unused export */
export { default as B } from './B'
```

and it is dropped after the optimization phase.

optimized webpack::lib/index.js
```js
export { default as A } from './A'
```

And since the export is dropped, the code of B is no longer included in the final bundle. Yay !

### Side effects ?

The problem is when B has a side effect, for example setting a global in window, inserting a CSS etc... In our case though, the CSS is extracted when transpiling. Here are our non transpiled files : 

B.styl
```
.toto
  background: red
```

B.jsx
```jsx
import styles from './B.styl'
```

The content from B.styl is extracted into a big CSS stylesheet. 
```css
/* A.styl */
/* blablabla */

/* B.styl */
.toto {
  background: red
}
```

and the import is replaced by an object containing the name of the classes in the transpiled JS.

transpiled:B.js
```
const styles = { toto: 'toto' }
```

So after transpilation, we have no references to CSS files in our JS files; the big CSS stylesheet is meant to included by the host application. This means that our transpiled JS files have no side-effects.

---

This leaves the case of importing stylus files when using the non transpiled version, but I do not think this is a problem since we are not in the case of re-exporting. Webpack will not flag exports as unused (no exports to flag). So the CSS code cannot be ignored by webpack.

The only problem we could have would be the order of insertion of the CSS when working with non transpiled files. B.jsx can be ignored, thuse B.styl can be ignored. If B.styl includes global styles, like `body { background: red }`, it could potentially be a problem if this global CSS is inserted earlier or later. I personally think that we will have few cases like that, we should not rely on CSS order (inter-files) and we should not have global CSS rules in components styl.

Reply:
Thanks for the guide :+1: